src/sources/soundsourcemp3.cpp

```diff
@@ -109,7 +109,9 @@ inline bool isUnrecoverableError(mad_error error) {
     return (MAD_ERROR_NONE != error) && !MAD_RECOVERABLE(error);
 }
 
-inline bool hasUnrecoverableError(const mad_stream* pMadStream) {
+[[maybe_unused]] // only used in DEBUG_ASSERT
```

Comment:
Please hide this entirely from the compiler by using `#ifdef MIXXX_DEBUG_ASSERTIONS_ENABLED`
This way we will later get the unused warning if the function becomes truly unused. 
It also self explaining and does not required a comment.